### PR TITLE
#1498 Apply global JSON unmarshalling to Node APIs

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1498)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/NodeApi.scala [hunk 1]
```diff
@@ -29,10 +29,12 @@ object NodeApi {
...
 
   case class Update(port: Option[Int], user: Option[String], password: Option[String])
-  implicit val NODE_UPDATE_JSON_FORMAT: RootJsonFormat[Update] = jsonFormat3(Update)
+  implicit val NODE_UPDATE_JSON_FORMAT: RootJsonFormat[Update] =
+    JsonRefiner[Update].format(jsonFormat3(Update)).connectionPort("port").rejectEmptyString().refine
 
   case class Creation(name: String, port: Int, user: String, password: String) extends CreationRequest
-  implicit val NODE_CREATION_JSON_FORMAT: RootJsonFormat[Creation] = jsonFormat4(Creation)
+  implicit val NODE_CREATION_JSON_FORMAT: RootJsonFormat[Creation] =
+    JsonRefiner[Creation].format(jsonFormat4(Creation)).connectionPort("port").rejectEmptyString().refine
```
Comment (chia7712) [inline ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/NodeApi.scala]: Consider that the basic connection is based on ssh, and should we auto-fill the 22 for ignored port?

Comment (saivirtue) [reply ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/NodeApi.scala]: good idea. Let me try to use `keyAndDefaultNumber`

**Comment by saivirtue** [comment]
retry

**Comment by saivirtue** [comment]
retry

**Comment by saivirtue** [comment]
the failed manager tests are traced on #1609, merge this pr later.

**Pull request merged**